src/test/java/cubicchunks/world/generator/TestCubeGenerator.java

```diff
@@ -0,0 +1,71 @@
+/*
+ *  This file is part of Cubic Chunks Mod, licensed under the MIT License (MIT).
+ *
+ *  Copyright (c) 2015 contributors
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package cubicchunks.world.generator;
+
+import static org.junit.Assert.fail;
+
+import cubicchunks.api.ICubicWorldGenerator;
+import cubicchunks.worldgen.generator.CubeGeneratorsRegistry;
+import mcp.MethodsReturnNonnullByDefault;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraft.world.chunk.IChunkGenerator;
+import net.minecraft.world.chunk.IChunkProvider;
+import net.minecraftforge.fml.common.registry.GameRegistry;
+
+import org.junit.Test;
+
+import java.util.Random;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+public class TestCubeGenerator implements ICubicWorldGenerator {
+
+    boolean testPassed = false;
+
+    @Test
+    public void testGeneration() {
+        GameRegistry.registerWorldGenerator(this, 0);
+        try {
+            CubeGeneratorsRegistry.computeSortedGeneratorList();
+        } catch (Throwable e) {
+            fail("CubeGeneratorsRegistry failed on attempt to extract fields");
```

Comment:
no reason to catch and explicitly call fail(), it will fail when exception is thrown anyway